### PR TITLE
AI Block Mapping | Fix icon block ai mapping

### PR DESCRIPTION
**Pull request by shrutisuresh16**
Fix icon-block dual-icon rendering for productLockup + logo
- When both `productLockup` and `logo` are present, clone the lockup `<p>` into a second slot and render the logo below the app tile — the mapping config only provides one DOM target today
- Add `icon-area` class and strip fixed `width`/`height` on the cloned slot so Milo CSS caps logo size correctly (fixes oversized second icon)
- Add early `return` in `handleLogo` after anchor-based injection to avoid duplicate picture elements
- Add picture/img fallback in `handleLogo` for empty logo slots (edge case)

Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #<gh-issue-id>

Test URLs:
- Before: https://main--{repo}--{owner}.aem.live/
- After: https://<branch>--{repo}--{owner}.aem.live/

### CODE REVIEW

File: streamlibs/blocks/icon-block.js [hunk 1]
```diff
@@ -77,6 +77,16 @@ function handleAvatar(value, areaEl) {
   if (altText) imgEl.alt = altText;
 }
 
+function createLogoSlot(lockupArea) {
```
Comment (saurabhsircar11) [inline streamlibs/blocks/icon-block.js:80]: Hmm i think it will be better to edit the icon block template itself and add a logo placeholder instead of adding it through code here.
same as product lockup we can put a switch condition for logo and do it.

**Review by maagrawal16 — APPROVED** [review]
